lib/sam/common/gpio_common_3a3u3x.c

```diff
@@ -19,16 +32,20 @@
 
 #include <libopencm3/sam/gpio.h>
 
+
+/** @brief Initialize GPIO pins
+ *
+ * @param[in] port uint32_t: GPIO Port base address
+ * @param[in] pin uint32_t
+ * @param[in] flags enum gpio_flags
+ */
 void gpio_init(uint32_t port, uint32_t pins, enum gpio_flags flags)
 {
-	switch (flags & 3) {
+	switch (flags & 0x7) {
```

Comment:
thats just the mask for extracting the pin direction or connected peripheral from the enum. it changed, because on some devices are more peripheriy blocks than PERIPHA and PERIPHB, so we need 3 bits here